### PR TITLE
Change default handoffConditionTimeout to 15 minutes.

### PR DESCRIPTION
**Pull request by gianm**
Most of the time, when handoff is taking this long, it's because something is preventing Historicals from loading new data. In this case, we have two choices:

1) Stop making progress on ingestion, wait for Historicals to load stuff,
   and keep the waiting-for-handoff segments available on realtime tasks.
   (handoffConditionTimeout = 0, the current default)

2) Continue making progress on ingestion, by exiting the realtime tasks
   that were waiting for handoff. Once the Historicals get their act
   together, the segments will be loaded, as they are still there on
   deep storage. They will just not be continuously available.
   (handoffConditionTimeout > 0)

I believe most users would prefer [2], because [1] risks ingestion falling behind the stream, which causes many other problems. It can cause data loss if the stream ages-out data before we have a chance to ingest it.

Due to the way tuningConfigs are serialized -- defaults are baked into the serialized form that is written to the database -- this default change will not change anyone's existing supervisors. It will take effect for newly created supervisors.

### CODE REVIEW

**Comment by abhishekagarwal87** [comment]
Thinking a bit about this change and discussing it with @kfaraz, it probably is ok to make this change. Initially, I was worried about this leading to holes in the data and recovery. But since it's only the hand-off, those holes will be temporary since the segment has been created.

**Review by abhishekagarwal87 — APPROVED** [review]

File: docs/development/extensions-core/kafka-supervisor-reference.md [hunk 1]
```diff
@@ -204,7 +204,7 @@ The `tuningConfig` is optional and default parameters will be used if no `tuning
...
+| `handoffConditionTimeout`         | Long           | Milliseconds to wait for segment handoff. It must be >= 0, where 0 means to wait forever.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | no (default == 900000 [15 minutes])                                                                          |
```
Comment (ektravel) [inline docs/development/extensions-core/kafka-supervisor-reference.md:207]:
```suggestion
| `handoffConditionTimeout`         | Long           | Number of milliseconds to wait for segment handoff. Set to a value >= 0, where 0 means to wait indefinitely.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | no (default == 900000 [15 minutes])                                                                          |
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 2]
```diff
@@ -283,7 +283,7 @@ The `tuningConfig` is optional. If no `tuningConfig` is specified, default param
 |`indexSpec`|Object|Tune how data is indexed. See [IndexSpec](#indexspec) for more information.|no|
 |`indexSpecForIntermediatePersists`|Object|Defines segment storage format options to be used at indexing time for intermediate persisted temporary segments. This can be used to disable dimension/metric compression on intermediate segments to reduce memory required for final merging. However, disabling compression on intermediate segments might increase page cache use while they are used before getting merged into final segment published, see [IndexSpec](#indexspec) for possible values.| no (default = same as `indexSpec`)|
 |`reportParseExceptions`|Boolean|If true, exceptions encountered during parsing will be thrown and will halt ingestion; if false, unparseable rows and fields will be skipped.|no (default == false)|
-|`handoffConditionTimeout`|Long| Milliseconds to wait for segment handoff. It must be >= 0, where 0 means to wait forever.| no (default == 0)|
+|`handoffConditionTimeout`|Long| Milliseconds to wait for segment handoff. It must be >= 0, where 0 means to wait forever.| no (default == 900000 [15 minutes])|
```
Comment (ektravel) [inline docs/development/extensions-core/kinesis-ingestion.md:286]:
```suggestion
|`handoffConditionTimeout`|Long| Number of milliseconds to wait for segment handoff. Set to a value >= 0, where 0 means to wait indefinitely.| no (default == 900000 [15 minutes])|
```

**Review by ektravel — APPROVED** [review]
A couple of minor nits.

**Pull request merged**